public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts

```diff
@@ -1,4 +1,4 @@
-import { set } from 'lodash';
+import { set, uniq } from 'lodash';
```

Comment:
Some context for the changes in this file. Before we started the work to move resource requests to the new layered architecture, CloudWatch logs did not use the CallResource api. All requests - `StartQuery`, `GetQueryResult`, `StopQuery`, `DescribeLogGroups`, `DescribeAllLogGroups`,  `GetLogGroupFields`, `GetLogEvents` - went through the DataQuery api. Using the DataQuery api for discovery data is bad practice and it also made it hard to structure code both in backend and in the frontend. After this pr, the only discovery data request that still goes through the DataQuery  api is the GetLogEvents request, but hopefully that can be removed too soon. 

Previously, all the requests went through the `makeLogActionRequest` method. The calling method was passing the `makeReplacement` boolean arg, and if it was set to true, some(?) of the props passed would be replaced as template variables. This was very un-intuitive, so main goal here is to interpolate variables where the query is actually created so that it becomes easier to test and read the code.

Reply:
Yes good idea!